### PR TITLE
implemented some new methods in browser context

### PR DESCRIPTION
**Pull request by TMcreative1**

### CODE REVIEW

**Review by Roman-Mitusov — CHANGES_REQUESTED** [review]
Please make the appropriate changes according to review comments

File: src/main/kotlin/com/playwright/remote/engine/waits/TimeoutSettings.kt [hunk 1]
```diff
@@ -1,4 +1,43 @@
 package com.playwright.remote.engine.waits
 
-class TimeoutSettings {
+import com.playwright.remote.engine.waits.api.IWait
+import com.playwright.remote.engine.waits.impl.WaitNever
+import com.playwright.remote.engine.waits.impl.WaitTimeout
+
+class TimeoutSettings(_parent: TimeoutSettings?) {
```
[the remark this answers is not in the document]
Comment (TMcreative1) [reply src/main/kotlin/com/playwright/remote/engine/waits/TimeoutSettings.kt]: Thanks, done

File: src/main/kotlin/com/playwright/remote/engine/waits/TimeoutSettings.kt [hunk 2]
```diff
@@ -1,4 +1,43 @@
...
+    private val defaultTimeoutMs = (30_000).toDouble()
+
+    private var parent: TimeoutSettings? = _parent
+    private var defaultTimeout = defaultTimeoutMs
+    private var defaultNavigationTimeout = defaultTimeoutMs
+
+    fun setDefaultTimeout(timeout: Double) {
```
[the remark this answers is not in the document]
Comment (TMcreative1) [reply src/main/kotlin/com/playwright/remote/engine/waits/TimeoutSettings.kt]: Thanks, done

File: src/main/kotlin/com/playwright/remote/engine/options/WaitForPageOptions.kt [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
 
-class WaitForPageOptions {
+import com.playwright.remote.engine.page.api.IPage
+import java.util.function.Predicate
+
+class WaitForPageOptions(
+    /**
+     * Receives the {@code Page} object and resolves to truthy value when the waiting should resolve.
+     */
+    var predicate: Predicate<IPage>? = null,
```
[the remark this answers is not in the document]
Comment (TMcreative1) [reply src/main/kotlin/com/playwright/remote/engine/options/WaitForPageOptions.kt]: Thanks, done

**Review by Roman-Mitusov — APPROVED** [review]
Ready to be merged

**Pull request merged**